### PR TITLE
feat: add support for `.http_uri` and `.ws_uri` in ProviderAPI v0.6.19 [APE-1381]

### PR DESCRIPTION
**Pull request by fubuloubu**
### What I did

fixes: #

### Checklist

- [ ] Passes all linting checks (pre-commit and CI jobs)
- [ ] New test cases have been added and are passing
- [ ] Documentation has been updated
- [ ] PR title follows [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) standard (will be automatically included in the changelog)

### CODE REVIEW

**Review by antazoey — APPROVED** [review]

File: ape_infura/provider.py [hunk 1]
```diff
@@ -48,6 +48,16 @@ def uri(self) -> str:
...
 
+    @property
+    def http_uri(self) -> str:
+        # NOTE: Overriding `Web3Provider.http_uri` implementation
+        return self.uri
+
+    @property
+    def ws_uri(self) -> str:
+        # NOTE: Overriding `Web3Provider.ws_uri` implementation
+        return "ws" + self.uri[4:]  # Remove `http` in default URI w/ `ws`
```
Comment (fubuloubu) [inline ape_infura/provider.py:59]: Forgot that this needs the update to add ws to the uri in the middle

**Review by antazoey — DISMISSED** [review]

File: ape_infura/provider.py (hunk 1 above)
Comment (fubuloubu) [reply ape_infura/provider.py:59]:
```suggestion
        # Remove `http` in default URI w/ `ws`, also infura adds `/ws` to URI
        return "ws" + self.uri[4:].replace("v3", "ws/v3")
```

**Comment by fubuloubu** [comment]
Note: probably should add tests to this and the Alchemy update to ensure the ws/http uris function

**Comment by antazoey** [comment]
> Note: probably should add tests to this and the Alchemy update to ensure the ws/http uris function

Test added

**Review by antazoey — APPROVED** [review]

**Pull request merged**